### PR TITLE
incremental symbols

### PR DESCRIPTION
**Pull request by tjdevries**
i don't know why I suddenly pushed to a different branch, but here it is.

@chrismwendt 

We might need to [`VACUUM`](https://www.sqlite.org/lang_vacuum.html) every once in a while, but it's not clear right now that it's worth it. More rows get inserted than deleted over time, so I would expect SQLite to eventually reuse the empty space.

Resolves https://github.com/sourcegraph/sourcegraph/issues/27611

### CODE REVIEW

File: cmd/symbols/internal/symbols/search.go [hunk 1]
```diff
@@ -112,14 +117,44 @@ func (s *Service) search(ctx context.Context, args protocol.SearchArgs) (*result
...
-				log15.Error("Unable to parse repository symbols within the context", "repo", args.Repo, "commit", args.CommitID, "query", args.Query)
-			}
 			return err
 		}
+
+		start := time.Now()
+
+		fmt.Println("================================================================================")
+		if newest == "" {
+			fmt.Println("GOTTA MAKE A NEW GUY")
```
Comment (efritz) [inline cmd/symbols/internal/symbols/search.go:130]: Maybe remove these pre-merge?

File: cmd/symbols/main.go [hunk 2]
```diff
@@ -66,8 +67,48 @@ func main() {
...
+				case 'A':
+					changes.Added = append(changes.Added, path)
+				case 'M':
+					changes.Modified = append(changes.Modified, path)
+				case 'D':
+					changes.Deleted = append(changes.Deleted, path)
+				}
+			}
+
+			fmt.Println("Changes A M D", len(changes.Added), len(changes.Modified), len(changes.Deleted))
```
Comment (efritz) [inline cmd/symbols/main.go:109]: Same here (and a few other files).

**Review by efritz — APPROVED** [review]

File: cmd/symbols/internal/symbols/search.go [hunk 3]
```diff
@@ -350,22 +394,85 @@ func (s *Service) writeAllSymbolsToNewDB(ctx context.Context, dbFile string, rep
...
+	}
+
+	// Write new commit
+	_, err = tx.Exec(`UPDATE meta SET revision = ?`, string(commitID))
+	if err != nil {
+		return err
+	}
+
+	// git diff
+	changes, err := s.GitDiff(ctx, repoName, oldCommit, commitID)
```
Comment (tjdevries) [inline cmd/symbols/internal/symbols/search.go:445]: I think this should be `GitDiffNames` or something, this name is not descriptive enough IMO

File: cmd/symbols/internal/symbols/search.go [hunk 4]
```diff
@@ -384,3 +491,66 @@ func SanityCheck() error {
 
 	return nil
 }
+
+// findNewestFile lists the directory and returns the newest file's path, prepended with dir.
+func findNewestFile(dir string) (string, error) {
```
Comment (tjdevries) [inline cmd/symbols/internal/symbols/search.go:496]: name is not probably descriptive enough now because it only finds the ".zip" files

**Pull request merged**